app/models/station.rb

```diff
@@ -0,0 +1,77 @@
+class Station < ActiveRecord::Base
+  validates :name, :dock_count, :city_id, :installation_date, presence: true
+  has_many :trip_starts, :class_name => 'Trip', :foreign_key => 'start_station_id'
+  has_many :trip_ends, :class_name => 'Trip', :foreign_key => 'end_station_id'
+  belongs_to :city
+  
+
+  def self.total_stations
+    count
+  end
+
+  def self.average_bikes_per_station
+    average :dock_count
+  end
+
+  def self.max_bikes
+    maximum :dock_count
+  end
+
+  def self.station_with_max_bikes
+    where(dock_count: maximum(:dock_count))
+  end
+
+  def self.min_bikes
+    minimum :dock_count
+  end
+
+  def self.station_with_min_bikes
+    where(dock_count: minimum(:dock_count))
+  end
+
+  def self.newest
+    where(installation_date: maximum(:installation_date))
+  end
+
+  def self.oldest
+    where(installation_date: minimum(:installation_date))
+  end
+
+  def self.ride_starts(station_id)
+    find(station_id).trip_starts.count
+  end
+
+  def self.ride_ends(station_id)
+    find(station_id).trip_ends.count
+  end
+
+  def self.most_frequent_destination(station_id)
```

Comment:
if we create a class method, but we're passing in the primary key of a specific station, why not make this an instance method? 

that way you won't need to find the object before calculating some logic/querying.

let me know if this makes sense!